### PR TITLE
Add initial widget

### PR DESCRIPTION
**Pull request by PhilippKuhlmay**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Changelog TYPO3 Crawler widget
+
+## Version 1.0.0
```
Comment (tomasnorre) [inline CHANGELOG.md:3]: I would start with a 0.0.1 version 1.0.0 indicates that it's stable already. I would wait a little with "claiming" that.

File: Classes/Widgets/Provider/QueueSizeDataProvider.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FriendsOfCrawler\CrawlerWidgets\Widgets\Provider;
```
Comment (tomasnorre) [inline Classes/Widgets/Provider/QueueSizeDataProvider.php:5]: repository is crawler-widget, so would keep the namespace singular too.

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "friendsofcrawler/typo3-crawler-widget",
+  "type": "typo3-cms-extension",
+  "description": "Widgets for crawler extension",
+  "homepage": "https://github.com/tomasnorre/typo3-crawler-widget",
+  "support": {
+    "issues": "https://github.com/tomasnorre/typo3-crawler-widget/issues"
+  },
+  "license": "GPL-3.0",
```
Comment (tomasnorre) [inline composer.json:9]: This doesn't match with the licens information in the PHP Files or the LICENSE file. 

Composer say: GPL3
PHP: GPL2+
LICENSE: MIT

We need to clean that up :)

File: Documentation/Widgets/Index.rst [hunk 4]
```diff
@@ -0,0 +1,29 @@
+.. include:: ../Includes.txt
+
+=======
+Widgets
+=======
+
+.. versionadded:: 1.0.0
```
Comment (tomasnorre) [inline Documentation/Widgets/Index.rst:7]: Version similar to in CHANGELOG

**Review by tomasnorre — CHANGES_REQUESTED** [review]
The Documentation/.gitkeep isn't needed as there are files in the Documentation-folder.

File: composer.json (hunk 3 above)
Comment (PhilippKuhlmay) [reply composer.json:9]: Do you have any suggestion? I am not really familiar with all those licences.

File: Documentation/Widgets/Index.rst (hunk 4 above)
Comment (PhilippKuhlmay) [reply Documentation/Widgets/Index.rst:7]: I changed the version to 0.0.1

File: CHANGELOG.md (hunk 1 above)
Comment (PhilippKuhlmay) [reply CHANGELOG.md:3]: I changed the version to 0.0.1

File: Classes/Widgets/Provider/QueueSizeDataProvider.php (hunk 2 above)
Comment (PhilippKuhlmay) [reply Classes/Widgets/Provider/QueueSizeDataProvider.php]: I changed it to FriendsOfCrawler\CrawlerWidget

**Comment by PhilippKuhlmay** [comment]
> The Documentation/.gitkeep isn't needed as there are files in the Documentation-folder.

I deleted it. The gitkeep was a leftover from the commits where I created the Documentation folder without content.

File: composer.json (hunk 3 above)
Comment (tomasnorre) [reply composer.json:9]: GPL2+ or MIT, GPL3 is not compliant with the TYPO3 Core.

Comment (PhilippKuhlmay) [reply composer.json]: I decided to use GPL2+

File: Documentation/Index.rst [hunk 5]
```diff
@@ -0,0 +1,30 @@
+.. include:: Includes.txt
```
Comment (brotkrueml) [inline Documentation/Index.rst:1]: I suggest to use the "official" documentation template:
https://github.com/TYPO3-Documentation/TYPO3CMS-Example-ExtensionManual/blob/master/Documentation/Index.rst

File: Documentation/Widgets/Index.rst [hunk 6]
```diff
@@ -0,0 +1,29 @@
+.. include:: ../Includes.txt
+
+=======
+Widgets
+=======
+
+.. versionadded:: 0.0.1
```
Comment (brotkrueml) [inline Documentation/Widgets/Index.rst:7]: As the first release of the extension has this widget included, I think, the version information is not necessary.

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+# Changelog TYPO3 Crawler widget
+
+## Version 0.0.1
```
Comment (brotkrueml) [inline CHANGELOG.md:3]: Personally I prefer "Unreleased" and adjust it accordingly to the version when publishing the extension. That makes it clearer that the features are unreleased yet.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,2 @@
 # typo3-crawler_widget
-A widget for the TYPO3 Crawler that would give TYPO3 Backend users an easier overview of potential problems with the Crawler
+A collection of widgets for the TYPO3 Crawler that give TYPO3 Backend users an easier overview of potential problems with the Crawler.
```
Comment (brotkrueml) [inline README.md:2]: I suggest "... that give TYPO3 backend users information about the status of Crawler." - which is more neutral.

File: Resources/Private/Language/locallang.xlf [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes" ?>
+<xliff version="1.0">
+    <file source-language="en" datatype="plaintext" original="EXT:typo3-crawler-widget/Resources/Private/Language/locallang.xlf" date="2020-08-27T08:00:00Z"
```
Comment (brotkrueml) [inline Resources/Private/Language/locallang.xlf:3]: EXT:typo3-crawler-widget => EXT:crawler_widget (use the official TYPO3 extension key)

File: Resources/Private/Language/locallang.xlf [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+          product-name="typo3_crawler_widget">
```
Comment (brotkrueml) [inline Resources/Private/Language/locallang.xlf:4]: same here

File: composer.json [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  "autoload": {
+    "psr-4": {
+      "FriendsOfCrawler\\CrawlerWidget\\": "Classes/"
+    }
+  },
+  "config": {
+    "sort-packages": true
+  },
+  "minimum-stability": "dev",
```
Comment (brotkrueml) [inline composer.json:36]: Why "dev"?

File: ext_emconf.php [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+ * The TYPO3 project - inspiring people to share!
+ */
+
+$EM_CONF[$_EXTKEY] = [
+    'title' => 'TYPO3 Crawler Widgets',
+    'description' => 'Widgets for crawler extension',
+    'category' => 'be',
+    'constraints' => [
+        'depends' => [
+            'typo3' => '^10.4',
```
Comment (brotkrueml) [inline ext_emconf.php:24]: 'typo3' => '10.4.0-10.4.99' - etc.

File: docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+version: '3'
```
Comment (brotkrueml) [inline docker-compose.yml:1]: Do we need this file?

File: Documentation/Widgets/Index.rst (hunk 6 above)
Comment (PhilippKuhlmay) [reply Documentation/Widgets/Index.rst:7]: I removed the "versionadded" information

File: README.md (hunk 8 above)
Comment (PhilippKuhlmay) [reply README.md]: I changed this.

File: composer.json (hunk 11 above)
Comment (PhilippKuhlmay) [reply composer.json:36]: I had this to test something and forgot to change it back

File: ext_emconf.php (hunk 12 above)
Comment (PhilippKuhlmay) [reply ext_emconf.php:24]: @brotkrueml @tomasnorre 
I changed the versions for the typo3 packages but I am not sure about the aoepeople/crawler package. Is 9.0.0-9.9.99 enough?

File: docker-compose.yml (hunk 13 above)
Comment (PhilippKuhlmay) [reply docker-compose.yml]: No. I had this for generating the documentation locally but this can be removed.

File: CHANGELOG.md (hunk 7 above)
Comment (PhilippKuhlmay) [reply CHANGELOG.md]: I changed this to `## Unreleased version 0.0.1`

**Review by brotkrueml — APPROVED** [review]
From my side it is mergable. There are some things, like documentation, which can be adjusted in a next step.

File: composer.json [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+  ],
+  "require": {
+    "php": "^7.2",
+    "typo3/cms-core": "^10.4",
+    "typo3/cms-dashboard": "^10.4",
+    "aoepeople/crawler": "^9.0"
+  }
+  ,
+  "require-dev": {
+    "roave/security-advisories": "dev-master"
```
Comment (tomasnorre) [inline composer.json:26]: I find it ok. But I see it more on the CI level of people setting it up. But it's not harming having it here. I would just do it differently.

File: LICENSE [hunk 15]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2020 Tomas Norre Mikkelsen
+Copyright (c) 2020 Philipp Kuhlmay
```
Comment (tomasnorre) [inline LICENSE:3]: The License info here and in the files doesn't match regarding naming, perhaps a LICENSE file should be enough, and then just reference it in the PHP files, I'm considering doing the same for the crawler, when I get all files and the same license. 

But here we have the chance to do it from the beginning.

**Review by tomasnorre — APPROVED** [review]

**Pull request merged**